[REPLCompletions] follow up #43865, keep input tuple type in `MethodCompletion`

aviatesk opened the pull request:
`MethodCompletion.tt` is being used by an external consumer like Juno or
VSCode's runtime completion features to get additional information like
return type, etc.

vtjnash commented on stdlib/REPL/src/REPLCompletions.jl:
```diff
@@ -617,7 +619,7 @@ function complete_methods!(out::Vector{Completion}, @nospecialize(funct), args_e
     for match in m
         # TODO: if kwargs_ex, filter out methods without kwargs?
-        push!(out, MethodCompletion(match.method))
+        push!(out, MethodCompletion(t_in, match.method))
```
```suggestion
        push!(out, MethodCompletion(match.spec_types, match.method))
```

aviatesk commented on stdlib/REPL/src/REPLCompletions.jl:
```diff
@@ -515,20 +517,20 @@ function get_type(T, found::Bool, default_any::Bool)
 
 # Method completion on function call expression that look like :(max(1))
-MAX_METHOD_COMPLETIONS = 40
+const MAX_METHOD_COMPLETIONS = Ref(40)
```
If we configure this value, it feels "better" if we can do `REPL.REPLCompletions.MAX_METHOD_COMPLETIONS[] = 100` instead of `@eval REPL.REPLCompletions MAX_METHOD_COMPLETIONS = 100`.

vtjnash replied:
It is not really configurable. It is a function argument though, if desired.

aviatesk replied:
Then we basically need to copy and paste the definitions of `complete_methods` etc. If it's not really supposed to be configured, we should probably make it `const`.

vtjnash replied:
Just because we make it less convenient to change, does not mean we have to make it `const`

vtjnash replied:
If people eventually do need to change it regularly, this will let them try out the idea, then add a contextual parameters object to this code to encapsulate any configurations of interest

aviatesk replied:
gotcha, let's just remove this change.

aviatesk committed:
[REPLCompletions] follow up #43865, keep input tuple type in `MethodCompletion`

`MethodCompletion.tt` is being used by an external consumer like Juno or
VSCode's runtime completion features to get additional information like
return type, etc.

Co-Authored-By: Jameson Nash <vtjnash@gmail.com>
